### PR TITLE
benchmark: refactoring and rbd

### PR DESCRIPTION
**Pull request by jan--f**
This is a fairly large PR adding rbd benchmarking and some urgently needed refactoring of a lot of benchmark code. No changes here should interfer with any component outside of the benchmark part of DeepSea.

Main areas of changes are:

- `srv/salt/ceph/benchmarks` addition of rbd orchestration file
- `srv/salt/ceph/rbd/benchmarks` necessary state files for rbd benchmarking
- `srv/modules/runners/benchmark.py` was heavily modified...I hope for the better.
- `srv/pillar/ceph/benchmark` was renamed to `srv/pillar/ceph/benchmarks` to keep consistency with the naming in other places

I also added a basic rbd job specification with the respective template. More of that and proper documentation will come soon.

### CODE REVIEW

**Comment by jan--f** [comment]
Fixes #48

**Comment by denisok** [comment]
also adding some README as in srv/salt/ceph/diagnose could be very useful.

**Comment by jan--f** [comment]
@denisok agreed, but that is a seperate issue (diagnose is another part of DeepSea). I will however review the benchmark docs...was on my list and slipped through the cracks.

File: srv/modules/runners/benchmark.py [hunk 1]
```diff
@@ -147,25 +163,70 @@ def run(**kwargs):
...
+
+    return True
+
+def cephfs(**kwargs):
+    """
+    Run cephfs benchmark jobs
+    """
+
+    client_glob = 'I@roles:client-cephfs and I@cluster:ceph'
+    if 'client_glob' in kwargs:
```
Comment (jschmid1) [inline srv/modules/runners/benchmark.py]: client_glob = kwargs.get('client_glob', None) 
will save you one line and allows you to set a default/fallback

File: srv/modules/runners/benchmark.py [hunk 2]
```diff
@@ -147,25 +163,70 @@ def run(**kwargs):
...
+            log.error(error)
+            raise error
+
+def rbd(**kwargs):
+    """
+    Run rbd benchmark job
+    """
+
+    client_glob = 'I@roles:client-rbd and I@cluster:ceph'
+    if 'client_glob' in kwargs:
```
Comment (jschmid1) [inline srv/modules/runners/benchmark.py]: client_glob = kwargs.get('client_glob', None) 
same here

File: srv/modules/runners/benchmark.py (hunk 1 above)
Comment (jschmid1) [inline srv/modules/runners/benchmark.py]: sorry, it's
client_glob = kwargs.get('client_glob', client_glob)

File: srv/modules/runners/benchmark.py (hunk 2 above)
Comment (jan--f) [inline srv/modules/runners/benchmark.py]: good point...fixed

**Comment by swiftgist** [comment]
Moving this back to Next since other items are more pressing.  The branch will still exist.

**Pull request closed without merging**